spec/ddtrace/configuration/components_spec.rb

```diff
@@ -993,75 +986,187 @@
-          expect(components.profiler).to receive(:shutdown!) unless components.profiler.nil?
-          expect(components.runtime_metrics).to receive(:stop)
-            .with(true, close_metrics: false)
-          expect(components.runtime_metrics.metrics.statsd).to receive(:close)
-          expect(components.health_metrics.statsd).to receive(:close)
+        context 'when one of Statsd instances are reused' do
+          include_context 'replacement' do
+            let(:runtime_metrics_worker) { components.runtime_metrics }
+          end
 
-          shutdown!
+          it 'shuts down all components but the tracer' do
+            expect(components.tracer).to receive(:shutdown!)
+            expect(components.profiler).to receive(:shutdown!) unless components.profiler.nil?
+            expect(components.runtime_metrics).to receive(:stop)
+              .with(true, close_metrics: false)
+            expect(components.runtime_metrics.metrics.statsd).to_not receive(:close)
+            expect(components.health_metrics.statsd).to receive(:close)
+
+            shutdown!
+          end
         end
-      end
 
-      context 'when one of Statsd instances are reused' do
-        include_context 'replacement' do
-          let(:runtime_metrics_worker) { components.runtime_metrics }
+        context 'when both Statsd instances are reused' do
+          include_context 'replacement' do
+            let(:runtime_metrics_worker) { components.runtime_metrics }
+            let(:health_metrics) { components.health_metrics }
+          end
+
+          it 'shuts down all components but the tracer' do
+            expect(components.tracer).to receive(:shutdown!)
+            expect(components.profiler).to receive(:shutdown!) unless components.profiler.nil?
+            expect(components.runtime_metrics).to receive(:stop)
+              .with(true, close_metrics: false)
+            expect(components.runtime_metrics.metrics.statsd).to_not receive(:close)
+            expect(components.health_metrics.statsd).to_not receive(:close)
+
+            shutdown!
+          end
         end
+      end
+    end
 
-        it 'shuts down all components but the tracer' do
-          expect(components.tracer).to receive(:shutdown!)
-          expect(components.profiler).to receive(:shutdown!) unless components.profiler.nil?
+    context 'with metrics disabled' do
+      let(:runtime_metrics_enabled) { false }
+      let(:health_metrics_enabled) { false }
+
+      context 'given no replacement' do
+        let(:replacement) { nil }
+
+        it 'shuts down all components' do
           expect(components.runtime_metrics).to receive(:stop)
             .with(true, close_metrics: false)
-          expect(components.runtime_metrics.metrics.statsd).to_not receive(:close)
-          expect(components.health_metrics.statsd).to receive(:close)
+          expect(components.runtime_metrics.metrics.statsd).to be_nil
+          expect(components.health_metrics.statsd).to be_nil
```

Comment:
This... is a somewhat weird expectation: This is testing behavior of `Metrics`, not really of components...?

I think this test should be rephrased as a "it correctly handles nil statsd instances" and should not have these expectations here (I believe they belong in the `Metrics` specs).